### DISCUSSION TITLE
A way to bypass the GFW for Wireguard MCI

### DISCUSSION THREAD

**Discussion by amir-devman** [XTLS/Xray-core#2702] [General]
I've recently discovered something about Wireguard on MCI that I could connect and have no limitation for speed/latency.
Before I say how, I must say that Wireguard in my area at least for MCI is completely blocked and there's no way that you can connect directly, but on MTN which is the second biggest in Iran, you can connect to Wireguard.
So, I have two sim cards, one MCI and one MTN, I used NekoBox to connect to the Wireguard on MTN then I switched my data connection to MCI without turning off the NekoBox connection and my connection was still going and I had no limitation while I was connecting to a Wireguard server from MCI and it was super stable.
Long story short, after handshake you can have a stable connection.

If there was a way to do a handshake or maybe fragmented handshake to bypass it. I think this would be a great step-up as in Wireguard is a high-performance protocol. _I might be dumb saying this suggestion but I say what I think can work_

I'd suggest you test this and leave your thoughts below and might work for you as well. Maybe someone could bypass this or something. I hope this helps

**Comment by ashad765** [comment]
It seems like SNI blocking on MCI. You can change subdomain or domain and check again. You can also try other udp based protocol like tuic and hy2. In theory tuic and hy2 should be faster than wireguard on unstable network.

**Comment by shakibamoshiri** [comment]
> I've recently discovered something about Wireguard on MCI that I could connect and have no limitation for speed/latency. Before I say how, I must say that Wireguard in my area at least for MCI is completely blocked and there's no way that you can connect directly, but on MTN which is the second biggest in Iran, you can connect to Wireguard. So, I have two sim cards, one MCI and one MTN, I used NekoBox to connect to the Wireguard on MTN then I switched my data connection to MCI without turning off the NekoBox connection and my connection was still going and I had no limitation while I was connecting to a Wireguard server from MCI and it was super stable. Long story short, after handshake you can have a stable connection.
> 
> If there was a way to do a handshake or maybe fragmented handshake to bypass it. I think this would be a great step-up as in Wireguard is a high-performance protocol. _I might be dumb saying this suggestion but I say what I think can work_
> 
> I'd suggest you test this and leave your thoughts below and might work for you as well. Maybe someone could bypass this or something. I hope this helps

Testing with WireGuard client and seems to work .  
do not know is done by the client/server or both but seems the handshake is held (kept) even after network changes so wg client do not retry connecting and uses prior handshake .

This could be a great feature for other VPN server/clients so to start with a non-blocked net then switching to the main net which blocked that protocol.

One other note is that WG uses UDP not TCP.
